doc/hardware/porting/board_porting.rst

```diff
@@ -310,33 +447,57 @@ application for it.
 Setting Kconfig configuration values is documented in detail in
 :ref:`setting_configuration_values`.
 
-There are three mandatory Kconfig files in the board directory for a board
-named ``plank``:
+There is one mandatory Kconfig file in the board directory, and several optional
+files for a board named ``plank``:
 
 .. code-block:: none
 
-   boards/<ARCH>/plank
-   ├── Kconfig.board
+   boards/<vendor>/plank
+   ├── Kconfig
+   ├── Kconfig.plank
    ├── Kconfig.defconfig
-   └── plank_defconfig
+   ├── plank_defconfig
+   └── plank_<identifier>_defconfig
+
+:file:`Kconfig.plank`
+  A shared Kconfig file which can be sourced both in Zephyr Kconfig and sysbuild
+  Kconfig trees.
 
-:file:`Kconfig.board`
-  Included by :zephyr_file:`boards/Kconfig` to include your board
-  in the list of options.
+  This file selects the SoC in the Kconfig tree and potential other SoC related
+  Kconfig settings. This file must not select anything outside the re-usable
+  Kconfig board and SoC trees.
 
-  This should at least contain a definition for a ``BOARD_PLANK`` option,
-  which looks something like this:
+  A :file:`Kconfig.plank` may look like this:
 
   .. code-block:: kconfig
 
      config BOARD_PLANK
-     	bool "Plank board"
-     	depends on SOC_SERIES_YOUR_SOC_SERIES_HERE
-     	select SOC_PART_NUMBER_ABCDEFGH
+     	select SOC_SOC1
+
+  The Kconfig symbols :kconfig:option:`BOARD_<board>` and
+  :kconfig:option:`BOARD_<board_with_identifier>` are constructed by the build
+  system, therefore no type shall be defined in above code snippet.
+
+:file:`Kconfig`
+  Included by :zephyr_file:`boards/Kconfig`.
+
+  This file can add Kconfig settings which are specific to the current board.
+
+  Not all boards have a :file:`Kconfig` file.
+
+  An example could be heap size adjustment.
+
+  .. code-block:: kconfig
+
+     config HEAP_MEM_POOL_ADD_SIZE_BOARD
+     	int
+     	default <value> if <condition>
```

Comment:
Now, this one looks like it belongs in `Kconfig.defconfig`, not `Kconfig`.
(I'm concerned about it being hard for users to decide what goes in which file.)

What I had in mind was something like `config BOARD_ENABLE_DCDC`, as an example of a board-specific setting that would appear in the "Board Options" menu.

Reply:
Rephrased, and included some guideline as well.

Reply:
> Now, this one looks like it belongs in `Kconfig.defconfig`, not `Kconfig`.

actually, this should be in `Kconfig`, not `Kconfig.defconfig` because it's defining a new Kconfig symbol, `HEAP_MEM_POOL_ADD_SIZE_BOARD` of type int. It's not adjusting the default for an existing setting (which is the purpose of Kconfig.default).